osu.Game/Input/Bindings/GlobalActionContainer.cs

```diff
@@ -84,34 +84,45 @@ public static IEnumerable<GlobalAction> GetGlobalActionsFor(GlobalActionCategory
+                    new KeyBinding(InputKey.Down, GlobalAction.SelectNext),
 
-            new KeyBinding(InputKey.Left, GlobalAction.SelectPreviousGroup),
-            new KeyBinding(InputKey.Right, GlobalAction.SelectNextGroup),
+                    new KeyBinding(InputKey.Left, GlobalAction.SelectPreviousGroup),
+                    new KeyBinding(InputKey.Right, GlobalAction.SelectNextGroup),
 
-            new KeyBinding(InputKey.Space, GlobalAction.Select),
-            new KeyBinding(InputKey.Enter, GlobalAction.Select),
-            new KeyBinding(InputKey.KeypadEnter, GlobalAction.Select),
+                    new KeyBinding(InputKey.Space, GlobalAction.Select),
+                    new KeyBinding(InputKey.Enter, GlobalAction.Select),
+                    new KeyBinding(InputKey.KeypadEnter, GlobalAction.Select),
 
-            new KeyBinding(InputKey.Escape, GlobalAction.Back),
-            new KeyBinding(InputKey.ExtraMouseButton1, GlobalAction.Back),
+                    new KeyBinding(InputKey.Escape, GlobalAction.Back),
+                    new KeyBinding(InputKey.ExtraMouseButton1, GlobalAction.Back),
 
-            new KeyBinding(new[] { InputKey.Alt, InputKey.Home }, GlobalAction.Home),
+                    new KeyBinding(new[] { InputKey.Alt, InputKey.Home }, GlobalAction.Home),
 
-            new KeyBinding(new[] { InputKey.Control, InputKey.Shift, InputKey.F }, GlobalAction.ToggleFPSDisplay),
-            new KeyBinding(new[] { InputKey.Control, InputKey.T }, GlobalAction.ToggleToolbar),
-            new KeyBinding(new[] { InputKey.Control, InputKey.Shift, InputKey.S }, GlobalAction.ToggleSkinEditor),
+                    new KeyBinding(new[] { InputKey.Control, InputKey.Shift, InputKey.F }, GlobalAction.ToggleFPSDisplay),
+                    new KeyBinding(new[] { InputKey.Control, InputKey.T }, GlobalAction.ToggleToolbar),
+                    new KeyBinding(new[] { InputKey.Control, InputKey.Shift, InputKey.S }, GlobalAction.ToggleSkinEditor),
 
-            new KeyBinding(new[] { InputKey.Control, InputKey.Alt, InputKey.R }, GlobalAction.ResetInputSettings),
+                    new KeyBinding(new[] { InputKey.Control, InputKey.Alt, InputKey.R }, GlobalAction.ResetInputSettings),
 
-            new KeyBinding(new[] { InputKey.Control, InputKey.Shift, InputKey.R }, GlobalAction.RandomSkin),
+                    new KeyBinding(new[] { InputKey.Control, InputKey.Shift, InputKey.R }, GlobalAction.RandomSkin),
 
-            new KeyBinding(InputKey.F10, GlobalAction.ToggleGameplayMouseButtons),
-            new KeyBinding(InputKey.F12, GlobalAction.TakeScreenshot),
-        };
+                    new KeyBinding(InputKey.F10, GlobalAction.ToggleGameplayMouseButtons),
+                    new KeyBinding(InputKey.F12, GlobalAction.TakeScreenshot),
+                };
+
+                if (OperatingSystem.IsWindows())
+                    bindings.Append(new KeyBinding(InputKey.Insert, GlobalAction.BossKey));
```

Comment:
This is the LINQ extension `Append`, right? It returns an enumerable with the item appended to it, it doesn't mutate the array...

Reply:
yeah, it's the LINQ version. the array has a fixed size, i can't append an element without recreating one with a different size. there's no non-LINQ append method.

Reply:
As in right now you're appending the boss key and throwing the appended enumerable away, you're supposed to do `bindings = bindings.Append(...)` or `return bindings;`

Reply:
oh yeah how did i miss that